oem/ibm/libpldmresponder/utils.cpp

```diff
@@ -195,18 +195,47 @@ int writeToUnixSocket(const int sock, const char* buf, const uint64_t blockSize)
 
 Json convertBinFileToJson(const fs::path& path)
 {
+    if (!fs::exists(path))
+    {
+        error("convertBinFileToJson: file does not exist: {PATH}", "PATH",
+              path);
+        return Json{};
+    }
     std::ifstream file(path, std::ios::in | std::ios::binary);
     std::streampos fileSize;
-
-    // Get the file size
+    if (!file.is_open())
+    {
+        error("convertBinFileToJson: failed to open file: {PATH}", "PATH",
+              path);
+        return Json{};
+    }
     file.seekg(0, std::ios::end);
     fileSize = file.tellg();
     file.seekg(0, std::ios::beg);
 
-    // Read the data into vector from file and convert to json object
+    if (fileSize <= 0)
+    {
+        error(
+            "convertBinFileToJson: file {PATH} is empty or invalid size {SIZE}",
+            "PATH", path, "SIZE", fileSize);
+        return Json{};
+    }
     std::vector<uint8_t> vJson(fileSize);
     file.read((char*)&vJson[0], fileSize);
-    return Json::from_bson(vJson);
+
+    Json result;
+    try
+    {
+        result = Json::from_bson(vJson);
+    }
+    catch (const std::exception& e)
+    {
+        error(
+            "convertBinFileToJson: JSON::from_bson failed for {PATH}, error={ERR}",
+            "PATH", path, "ERR", e.what());
```

Comment:
follow the latest lg2 pattern - https://github.com/openbmc/docs/blob/master/anti-patterns.md#ill-defined-data-structuring-in-lg2-message-strings